core/src/main/java/org/commonjava/indy/core/content/DefaultDirectContentAccess.java

```diff
@@ -40,21 +49,63 @@
     @Inject
     private DownloadManager downloadManager;
 
+    @Inject
+    @WeftManaged
+    @ExecutorConfig( named = "direct-content-access" )
+    private ExecutorService executorService;
+
+
     public DefaultDirectContentAccess(){}
 
-    public DefaultDirectContentAccess( DownloadManager downloadManager )
+    public DefaultDirectContentAccess( final DownloadManager downloadManager, ExecutorService executorService )
     {
         this.downloadManager = downloadManager;
+        this.executorService = executorService;
     }
 
+
+    @Override
     public List<Transfer> retrieveAllRaw( final List<? extends ArtifactStore> stores, final String path,
                                           final EventMetadata eventMetadata )
             throws IndyWorkflowException
     {
-        final List<Transfer> txfrs = new ArrayList<Transfer>();
+        Logger logger = LoggerFactory.getLogger( getClass() );
+        CompletionService<Transfer> executor = new ExecutorCompletionService<>( executorService );
         for ( final ArtifactStore store : stores )
         {
-            final Transfer txfr = retrieveRaw( store, path, eventMetadata );
+            logger.debug( "Requesting retrieval of {} in {}", path, store );
+            executor.submit( new Callable<Transfer>()
+            {
+                @Override
+                public Transfer call() throws IndyWorkflowException
+                {
+                    logger.trace( "Retrieving {} in {}", path, store );
+                    Transfer txfr = retrieveRaw( store, path, eventMetadata );
+                    logger.trace( "Transfer {} in {} retrieved", path, store );
+                    return txfr;
+                }
+            });
+        }
+
+        final List<Transfer> txfrs = new ArrayList<>( stores.size() );
+        for ( ArtifactStore store : stores )
+        {
+            Transfer txfr;
+            try
+            {
+                logger.trace( "Waiting for transfer of {} in {}", path, store );
+                txfr = executor.take().get();
```

Comment:
I think this returns the first one to complete, not the next in order of submission. That could still result in incorrect ordering. Hmm, maybe the CompletionService isn't helpful here, since we're waiting for all of them to complete (which means it's constrained to the slowest one). If we used a map of ArtifactStore -> Future we could iterate through the original stores parameter and add the Transfer instances to the result in proper order. 

This is something I can take care of today, though...I won't block this PR for that.